python/core/auto_generated/symbology/qgsmasksymbollayer.sip.in

```diff
@@ -0,0 +1,98 @@
+/************************************************************************
+ * This file has been generated automatically from                      *
+ *                                                                      *
+ * src/core/symbology/qgsmasksymbollayer.h                              *
+ *                                                                      *
+ * Do not edit manually ! Edit header and run scripts/sipify.pl again   *
+ ************************************************************************/
+
+
+
+
+
+class QgsMarkerMaskSymbolLayer : QgsMarkerSymbolLayer
```

Comment:
Also - what happens with masks in legends?

Reply:
> We need to add api to limit it being exposed as an option ONLY for vector layer renderers.

On GUI side, perhaps with something added to QgsSymbolWidgetContext ?

> Also - what happens with masks in legends?

If no mask painter is defined, it is rendered as a regular symbol layer.

Reply:
> If no mask painter is defined, it is rendered as a regular symbol layer.

Wouldn't this then show the big red buffer in the legend?

Reply:
Yes it will. Do you think it's a problem (apart from the color) ?

Reply:
Yes - I think it's an issue. What I would suggest is doing something like:

- adding a new flag to QgsRenderContext, something like " Is GUI preview only"
- only rendering the red mask in gui previews - for all other contexts where a mask isn't used (e.g. legends), just totally skip masks and don't render anything.

Then you could safely use point layers with masked symbols in legends and they'll look as expected.

Reply:
I've changed things so that only QgsSymbol::bigSymbolPreviewImage() will render the mask. QgsSymbol::asImage() and QgsSymbolLayerUtils::symbolLayerPreviewImage() do not.